fab3/filters.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package fab3
+
+import (
+	"encoding/hex"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/hyperledger/fabric-chaincode-evm/fab3/types"
+)
+
+type filterEntry interface {
+	LastAccessTime() time.Time
+	Filter(s *ethService) ([]interface{}, error)
+}
+
+type logsFilter struct {
+	logArgs         *types.GetLogsArgs
+	latestBlockSeen uint64
+	lastAccessTime  time.Time
+}
+
+func (f *logsFilter) LastAccessTime() time.Time {
+	return f.lastAccessTime
+}
+
+func (f *logsFilter) Filter(s *ethService) ([]interface{}, error) {
+	s.logger.Debug("lastblockseen before filtering", f.latestBlockSeen)
+	currentLatestBlockSeen, err := s.parseBlockNum("latest")
+	if err != nil {
+		return nil, err
+	}
+	s.logger.Debug("latest blockseen", currentLatestBlockSeen)
+
+	var l []interface{}
+	if f.latestBlockSeen < currentLatestBlockSeen {
```

Comment:
This should be `<=`

Otherwise it will still run GetLogs when we have already seen the currentLatestBlock causing the error: `"failed to run filter id: 2: fromBlock number greater than toBlock number"`

Reply:
test would be running the filter twice in a row?

Reply:
I'm not sure how I would cause this?